platform-api/che-core-api-machine/src/test/java/org/eclipse/che/api/machine/server/MachineManagerTest.java

```diff
@@ -0,0 +1,142 @@
+import org.eclipse.che.api.machine.shared.MachineStatus;
+import org.eclipse.che.api.machine.shared.Recipe;
+import org.eclipse.che.api.machine.shared.dto.RecipeMachineCreationMetadata;
+import org.eclipse.che.api.machine.shared.dto.recipe.MachineRecipe;
+import org.eclipse.che.commons.env.EnvironmentContext;
+import org.eclipse.che.commons.user.UserImpl;
+import org.mockito.Mock;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.mockito.Matchers.anyBoolean;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+
+/**
+ * @author Anton Korneta
+ */
+@Listeners(MockitoTestNGListener.class)
+public class MachineManagerTest {
+    private static final String MACHINE_LOGS_DIR               = "target/logs-dir";
+    private static final int    DEFAULT_MACHINE_MEMORY_SIZE_MB = 1000;
+
+    private MachineManager manager;
+
+    @Mock
+    private MachineInstanceProviders machineInstanceProviders;
+    @Mock
+    private MachineRegistry          machineRegistry;
+
+    @BeforeMethod
+    public void setUp() {
+        final SnapshotDao snapshotDao = mock(SnapshotDao.class);
+        final EventService eventService = mock(EventService.class);
+
+        manager = spy(new MachineManager(snapshotDao,
+                                         machineRegistry,
+                                         machineInstanceProviders,
+                                         MACHINE_LOGS_DIR,
+                                         eventService,
+                                         DEFAULT_MACHINE_MEMORY_SIZE_MB));
+
+        EnvironmentContext envCont = new EnvironmentContext();
+        envCont.setUser(new UserImpl("user", null, null, null, false));
+        EnvironmentContext.setCurrent(envCont);
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+        EnvironmentContext.reset();
+    }
+
+    @Test(expectedExceptions = MachineException.class)
+    public void shouldBeAnExceptionWhileValidatingDockerDisplayName() throws Exception {
+        final RecipeMachineCreationMetadata machineCreationMetadata = mock(RecipeMachineCreationMetadata.class);
+        final MachineRecipe machineRecipe = mock(MachineRecipe.class);
+        final InstanceProvider instanceProvider = mock(InstanceProvider.class);
+        final String type = "type";
+        final Set<String> recipeTypes = new HashSet<String>() {{
```

Comment:
Hmm, actually here you create an instance of anonymous class which is not good, you can simplify it:
<code>new HashSet<>(Collections.singleton(type));